File: sdks/php/src/Resources/PurchaseOrdersResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\PurchaseOrder;
use Puxbay\Models\PaginatedResponse;

class PurchaseOrdersResource extends BaseResource
{
    public function list(int $page = 1, ?string $status = null): PaginatedResponse
    {
        $endpoint = "purchase-orders/?page={$page}";
        if ($status) {
            $endpoint .= "&status={$status}";
        }
        $data = $this->get($endpoint);
        return PaginatedResponse::fromArray($data, fn($item) => PurchaseOrder::fromArray($item));
    }

    public function getPurchaseOrder(string $poId): PurchaseOrder
    {
        $data = $this->get("purchase-orders/{$poId}/");
        return PurchaseOrder::fromArray($data);
    }

    public function create(PurchaseOrder $po): PurchaseOrder
    {
        $data = $this->post('purchase-orders/', $po->toArray());
        return PurchaseOrder::fromArray($data);
    }

    public function update(string $poId, PurchaseOrder $po): PurchaseOrder
    {
        $data = $this->patch("purchase-orders/{$poId}/", $po->toArray());
        return PurchaseOrder::fromArray($data);
    }

    public function receive(string $poId, array $items): PurchaseOrder
    {
        $data = $this->post("purchase-orders/{$poId}/receive/", ['items' => $items]);
        return PurchaseOrder::fromArray($data);
    }
}

File: sdks/dotnet/src/Resources/PurchaseOrdersResource.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class PurchaseOrdersResource
    {
        private readonly PuxbayClient _client;
        public PurchaseOrdersResource(PuxbayClient client) { _client = client; }

        public async Task<PaginatedResponse<PurchaseOrder>> ListAsync(int page = 1, string status = null)
        {
            string query = $"purchase-orders/?page={page}";
            if (!string.IsNullOrEmpty(status)) query += $"&status={status}";
            return await _client.GetAsync<PaginatedResponse<PurchaseOrder>>(query);
        }

        public async Task<PurchaseOrder> GetAsync(string poId)
        {
            return await _client.GetAsync<PurchaseOrder>($"purchase-orders/{poId}/");
        }

        public async Task<PurchaseOrder> CreateAsync(PurchaseOrder po)
        {
            return await _client.PostAsync<PurchaseOrder>("purchase-orders/", po);
        }

        public async Task<PurchaseOrder> UpdateAsync(string poId, PurchaseOrder po)
        {
            return await _client.PatchAsync<PurchaseOrder>($"purchase-orders/{poId}/", po);
        }

        public async Task<PurchaseOrder> ReceiveAsync(string poId, List<Dictionary<string, object>> items)
        {
            return await _client.PostAsync<PurchaseOrder>($"purchase-orders/{poId}/receive/", new { items });
        }
    }
}
